sys/shell/shell.c

```diff
@@ -280,21 +320,31 @@ static inline void print_prompt(void)
 #endif
 }
 
+static const char *TOOLONG_MESSAGE = "shell: maximum line length exceeded";
+
 void shell_run(const shell_command_t *shell_commands, char *line_buf, int len)
 {
     print_prompt();
 
     while (1) {
         int res = readline(line_buf, len);
 
-        if (res == EOF) {
-            break;
-        }
-
-        if (!res) {
-            handle_input_line(shell_commands, line_buf);
+        switch (res) {
+            case EOF:
+                goto shell_run_exit;
```

Comment:
this is probably a style question, but I think goto is controversal enough to just ```return``` here.

Reply:
I'd rather have the procedure return at only one point. Goto is controversial for those who do not know how to use it correctly:  forwards and outwards jumps are OK.

Reply:
there are (better) places where `goto` has its (controversial) benefits - here it looks rather cosmetic (to me) and I would go for a simple `return;` as @kaspar030 suggested.

Reply:
I'm _not_ going to `return` from a inside a `case` inside a `while`.